### ISSUE TITLE
Feature request: full support for meta tagging

### ISSUE THREAD

**Issue by alan12345678** [gnumdk/lollypop#571]
NOT a bug!

Is it possible/could you/do you intend to include full support for "meta tags"??

Since using your (wonderful) player I've paid much more attention to both tagging in general as well as the use of meta tags.

The more I use this player and enjoy the natural way of browsing, adding and playing tracks and albums, the more I come to feel that one of the MOST important things it must do is provide more/better control over how the library/artists are displayed and grouped. My comments on grouping and tagging came out of thinking about how one might expect to find music arranged inside a record shop/music store. Take some examples:

I have a lot of albums by "Bob Marley". I also have a couple of  albums by "The Wailers". In my normal tagging I like to reflect the same artist credits as given on sites such as Discogs or Wikipedia. Thus within my "Bob Marley" library *in terms of folder/directory structure on disc* will be albums credited to:

"Bob Marley"
"Bob Marley & The Wailers"
"The Wailers"

Currently under lollypop the only way to ensure that these albums (or single tracks) appear in the same part of the library is to include the tag "album artist" as identical in each case i.e. Bob Marley. Failing this I have to scroll from "B" down to "W" to find related music. My library list is LONG!!

However, the use of correct meta tags would mean that for each album or track I can use the normal tag "artist" as any of the above values, completely omit the "album artist" tag and then include either the "albumsort" meta tag, or the "albumartistsort" meta tag and use the value "Bob Marley" as a grouping device while still keeping correct values for artist within specific albums.

This would also 'cure' the huge problem that currently exists with compilation albums - of which I have dozens and dozens. For example, I may have a four disc set of country music. Each track will list "artist" as the name of performer on each track, while "album artist" presently says "various artists". At present therefore, I would have to scroll down to "V" and look within the huge library of "various artists" (where, incidentally, the collections may or may not be correctly displayed).

However, this four CD compilation might actually have a name, say "Country Music Hits". Thus I could stick with my present "artist" "album artist" tags, but include a meta tag for "albumsort" e.g. "country music hits" and therefore have an album of "various artists" which I choose to find under "C" and not  "V".

Note that one small bug/annoyance at present is that where I have a single track (NOT a complete album) and which track has different values for "artist" and "album artist" (example: Yo-Yo Ma & The Silk Road Ensemble; Yo-Yo Ma), this one track show up twice in the artist list - one for each value.

This is a very good player already, but I think that consistent and predictable tag support is essential.

**Comment by d9h02f** [comment]
tl;dr: respect sort tags

**Comment by gnumdk** [comment]
> Currently under lollypop the only way to ensure that these albums (or single tracks) appear in the same part of the library is to include the tag "album artist" as identical in each case i.e. Bob Marley. Failing this I have to scroll from "B" down to "W" to find related music. My library list is LONG!!

If your album is tagged "Bob Marley; The Wailers", it will be shown at near "Bob Marley" albums

**Comment by gnumdk** [comment]
![exemple](https://cloud.githubusercontent.com/assets/3918824/14491894/d10f98cc-017b-11e6-89fa-a1ead59834e7.png)

Here an album from DJ Pone, I force lollypop to show it near Svinkels albums by using "Svinkels; DJ Pone" as album artists

**Comment by alan12345678** [comment]
I realise that someone kindly provided a tl;dr for my expansive comments
;0) but there is a reason for being so verbose.
 
I see it in this way:
 
As the user I am free to choose the player which best serves my needs.
 
As the developer, I assume that you'd like your player to become popular
and widely used. Maybe not?
 
It is of little interest to me to note that IF I were to somehow 'fudge'
my tagging, and follow arbitrary rules supplied by the developer, I
MIGHT be able to find my music collection sorted "approximately".
 
What use is that to me?
 
I wish to observe correct tagging standards. Who knows - maybe one day
you'll stop developing your player, or a system update might render it
broken. And I then find myself with a LARGE music library which I have
now tagged according to some arbitrary standards and which is now
largely useless in my next best player of choice.
 
So no - in the kindest way possible - I do not want "if you add multiple
tags to your music it might show up NEAR other artists". I want to see
albums or tunes able to be sorted and grouped according to MY precise
needs, not by standards that the developer has decided must be applied
but which would not be universally respected ;0) As the tl;dr kindly
summarised it: RESPECT SORT ORDER tags.
 
And this is to say nothing presently about the frankly atrocious
handling of this player with multiple album/artist compilations
 
:)
 
----- Original message -----

**Comment by gnumdk** [comment]
Lollypop handles compilations like clementine and amarok, don't see anything atrocious....

**Comment by gnumdk** [comment]
Multi artist tags is a standard

**Comment by gnumdk** [comment]
Will add support for album artist sort tag. (You can already user artist sort tag)
But I will not support this:
Album1 with artist Artist1 and Sortname OneArtist1
Album2 with artist Artist1 and Sortname TwoArtist1

This will kill lollypop performances for something broken IMO

**Comment by alan12345678** [comment]
Thank you - that's kind of you.
 
Just to be fully clear then, you are saying that it will not be possible
to support the full range of available tags? Therefore it will be
pointless to expect it to respect such things as "albumsort"
"albumartistsort" "totaldiscs" etc?
 
Others have commented that they see the support of compilations as "no
worse than" other longer term players. Perhaps my lollypop installation
is STILL not working as expected despite my using only full releases on
a fully updated Fedora 23? Not sure there's any way of checking.
 
On my installation there's a strange, rather hit and miss business of it
seeing explicitly as "compilations" just three discs, none of which are
in fact compilations. Most of the multiple artist sets end up being
displayed in the left hand list of artists, where each single track
(from a compilation) is displayed as a single entry under artist name.
Not sure there's much I can do about this.
 
 
----- Original message -----

**Comment by gnumdk** [comment]
Compilations are album with "No album artists" and different "artists". If you wan them not marked as a compilation, set album artists.

**Comment by gnumdk** [comment]
>Threfore it will be pointless to expect it to respect such things as "albumsort" "albumartistsort" "totaldiscs" etc?

Why would you want two artists sorted with a different string?

**Comment by alan12345678** [comment]
Since you asked:
Most single artist albums aren't problematic - they display how and where one would expect. Compilations on the other hand...

Suppose you have maybe 200 compilation albums. These tend to have both "artist" tag per song, and "album artist" tag as "various artists". This could simply be blank. Now suppose you have five compilation albums of soul music and maybe 10 compilations of salsa music. Perhaps you would like to find them listed as "Soul Music 1", "Soul Music 2" , and "Salsa 1", "Salsa 2" etc.

I don't see how this is presently achieved. If I omit "album artist" then my albums don't list/display as compilations at all - I simply have maybe 40 different songs by 40 different artists scattered throughout my artist list in the left hand pane. I thought the very point of meta tags is that I could:

Tag each individual track within a compilation with "title" [name of song] and "artist" [performer of song]
Tag "album artist" as "various artists" or leave blank
Use "album sort" tag as "Soul Music 1" or perhaps "album artist sort" as "soul music"

This would ensure that my compilation shows up at a specific place/label, rather than finding myself having to wade through a now very large "various artists" listing under "V" in the left hand pane.

In my installation (which is to say that this may not be common or as designed) it struggles to correctly place an album of duets. For example:

Carmen McRae. I want to view all of the albums on which she appears in a single entry. However, on one album the artist credits are "Carmen McRae and George Shearing". On another they are "Cal Tjader | Carmen McRae". I would much prefer to leave album artists as is, and include "album sort" tag to value of "Carmen McRae", rather than re-edit tags to remove artist information simply for the sake of having them grouped according to my intent vs player constraints.

**Comment by gnumdk** [comment]
>I don't see how this is presently achieved. If I omit "album artist" then my albums don't list/display as compilations at all - I simply have maybe 40 different songs by 40 different artists scattered throughout my artist list in the left hand pane.

Impossible, two solutions:
- You don't even test this
- You found a bug

**Comment by gnumdk** [comment]
![capture d ecran de 2016-04-14 22-20-15](https://cloud.githubusercontent.com/assets/3918824/14542447/41a80bb4-028f-11e6-8244-f9f344017897.png)

**Comment by alan12345678** [comment]
Wonderful. Please explain how you achieved this.

**Comment by alan12345678** [comment]
I have NO secondary display pane... only the first one.

**Comment by gnumdk** [comment]
Without genres, you have all compilations in the same view.

**Comment by gnumdk** [comment]
>Please explain how you achieved this.
Compilations are album with "No album artists" and different "artists". If you wan them not marked as a compilation, set album artists.

**Comment by alan12345678** [comment]
This is unclear. I have done a LOT!! of experiments. If I take a compilation album - let's suppose 20 artists - and use only "artist" and "title" tags, it does NOT display as a single complation but as 20 different artists. I have tried using the "compilation" meta tag but it makes no difference.

More importantly though, what are your tagging methods that produced that screenshot above?? I have genres on EVERY. SINGLE. tune I own, but it doesn't look anything like your display above.

**Comment by gnumdk** [comment]
You need to enable "Show genres" in settings

**Comment by gnumdk** [comment]
Marked as a compilation in lollypop
![capture d ecran de 2016-04-14 22-32-29](https://cloud.githubusercontent.com/assets/3918824/14542769/effb7e98-0290-11e6-80ce-b5bd1f513564.png)

**Comment by gnumdk** [comment]
Another tips, lollypop only album year... Track year is "original year" and isn't supported.
If you use album year as track year (and so have different years in your compilations), you will have different albums...
But you are wrong, this tag is about "album release date", not track.

**Comment by alan12345678** [comment]
Thanks.

Year?? I have no problem with dates.
